### PR TITLE
dmalloc: Add recipe for dmalloc. Version: 5.5.2

### PR DESCRIPTION
**Pull request by hundeboll**
Signed-off-by: Christian Sørensen <christian.braunersorensen@prevas.dk>

### CODE REVIEW

File: recipes/dmalloc/dmalloc_5.5.2.oe [hunk 1]
```diff
@@ -0,0 +1,60 @@
+DESCRIPTION = "Debug Malloc Library"
+HOMEPAGE = "http://dmalloc.com"
+LICENSE = "CC-BY-SA-3.0"
+
+SRC_URI = "\
+  http://dmalloc.com/releases/dmalloc-${PV}.tgz \
+  file://configure-pagesize-HACK.patch \
```
Comment (kimrhh) [inline recipes/dmalloc/dmalloc_5.5.2.oe]: ugh, this patch seems like a really bad idea :) could we try figuring out where it is from? same with the other patch

**Comment by kimrhh** [comment]
fails for mingw, COMPATIBLE_HOST_ARCHS is ok with me

File: recipes/dmalloc/dmalloc_5.5.2.oe (hunk 1 above)
Comment (hundeboll) [inline recipes/dmalloc/dmalloc_5.5.2.oe]: Fixes the same issue, but with a decent description and less intrusive:
http://crosstool-ng.org/hg/crosstool-ng/file/tip/patches/dmalloc/5.5.2/110-block_size.patch

The other patch is needed with libc/compilers that optimizes str(n)dup into a macro that must be undefined before special cased. Maybe I should update the patch with a comment?

Comment (kimrhh) [inline recipes/dmalloc/dmalloc_5.5.2.oe]: would be nice, thanks!

**Pull request merged**